lib/src/bridge/characteristics_mixin.dart

```diff
@@ -66,11 +77,16 @@ mixin CharacteristicsMixin on FlutterBLE {
           )
           .catchError((errorJson) =>
               Future.error(BleError.fromJson(jsonDecode(errorJson.details))))
-          .then(
-            (rawJsonValue) =>
-                _parseCharacteristicWithValueWithTransactionIdResponse(
-                    peripheral, rawJsonValue),
-          );
+          .then((rawValue) {
+            String rawJsonValue = "";
+            if (rawValue is String) {
+              rawJsonValue = rawValue;
+            }
+            return _parseCharacteristicWithValueWithTransactionIdResponse(
+              peripheral, 
+              rawJsonValue
+            );
+          });
 
   Future<void> writeCharacteristicForIdentifier(
```

Comment:
@mikolak 
OFF-TOPIC:
it looks like the response is swallowed here and it returns void instead of `Characteristic? response`
is this intentional?

Same in other write methods where the response is optionally available, all those should return `Future<CharacteristicsWithValue?>` as opposed to `Future<Characteristics>` right?

Or am I overlooking something? 🤔

Reply:
the last commit is to mitigate this. 
Pls recommend a better exception to throw on null value erroneously coming from native.
Thanks!